runtime/expr/agg/math.go

```diff
@@ -39,32 +48,46 @@ func (m *mathReducer) Consume(val *zed.Value) {
 }
 
 func (m *mathReducer) consumeVal(val *zed.Value) {
-	// A numerical reducer inherits the type of the first numeric
-	// value it sees and coerces all future instances of this value
-	// to this initial type.
-	if m.typ == nil {
-		m.typ = val.Type
-	}
-	if val.IsNull() {
-		return
+	var id int
+	if m.math != nil {
+		var err error
+		// XXX We're not using the value coercion parts of coerce.Pair here.
+		// Would be better if coerce had a function that just compared types
+		// and returned the type to coerce to.
+		id, err = m.pair.Coerce(&m.nilVal, val)
```

Comment:
No need for `nilVal` here. The `Coerce` arguments don't escape to the heap so the `zed.NewValue` return value goes on the stack. (You can see this in the output of `go build -gcflags=-m ./runtime/expr/agg` and `go tool objdump -S -s mathReducer.*consumeVal dist/zq`).
```suggestion
		id, err = m.pair.Coerce(val, zed.NewValue(m.math.typ(), nil))
```